### PR TITLE
Feat: Forgot Password

### PR DESCRIPTION
**Pull request by ogbanugot**

### CODE REVIEW

File: internal/email/templates/reset.password.html [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                            </p>
+
+                            <p class="issue-text">
+                                For any enquiry or complaint, you can reply to this email.
+                            </p>
+                        </div>
+
+                        <div class="center footer">
+                            <p>© <a href="https://getconvoy.io">Convoy</a></p>
+                            <p>A Cloud native Webhook Service</p>
```
Comment (jirevwe) [inline internal/email/templates/reset.password.html:116]:
```suggestion
                            <p>An OSS Webhook Service</p>
```

File: services/user_service.go [hunk 2]
```diff
@@ -202,3 +212,93 @@ func (u *UserService) UpdatePassword(ctx context.Context, data *models.UpdatePas
 
 	return user, nil
 }
+
+func (u *UserService) GeneratePasswordResetToken(ctx context.Context, baseURL string, data *models.ForgotPassword) error {
+	var resetToken string
```
Comment (jirevwe) [inline services/user_service.go:217]: why is reset token defined here?

**Review by jirevwe — CHANGES_REQUESTED** [review]

File: services/user_service.go (hunk 2 above)
Comment (ogbanugot) [reply services/user_service.go:217]: oh I'm using a variable because I pass it into the `sendPasswordResetEmail` function as well.

Comment (jirevwe) [reply services/user_service.go:217]: Yeah, but it's initialized on line 230 so there's not need to define it here

Comment (ogbanugot) [reply services/user_service.go:217]: okay I see your point.

**Review by Subomi — APPROVED** [review]

**Review by jirevwe — APPROVED** [review]

**Review by danvixent — APPROVED** [review]

**Pull request merged**